### ISSUE TITLE
Learn how to make commits

### ISSUE THREAD

**Issue by uo288245** [Arquisoft/wiq_es04a#318]
Made to laugh a little bit to relieve stress

**Comment by and1na** [comment]
I think this is a very good initiative, some things I'm seeing these days lead me to think some unnecessary changes could lead to errors.

**Comment by PabloGOP** [comment]
Sorry but I am not getting it, I think this repo has more commits than any other in the subject so I'd say that we crack at making commits...

**Comment by uo288245** [comment]
> Sorry but I am not getting it, I think this repo has more commits than any other in the subject so I'd say that we crack at making commits...

i think that this is a great example to know how to make it:

![Image](https://github.com/Arquisoft/wiq_es04a/assets/145291757/7c2a7a82-f8c3-4b28-bb6a-9fb7548a2efd)

**Comment by uo288245** [comment]
> > Sorry but I am not getting it, I think this repo has more commits than any other in the subject so I'd say that we crack at making commits...
> 

even though this is the best one in the repo:


![Image](https://github.com/Arquisoft/wiq_es04a/assets/145291757/71743566-51ae-4789-92f3-18e68088637f)

**Comment by marco-qg** [comment]
# You can always learn from @alegarman2002
![image](https://github.com/Arquisoft/wiq_es04a/assets/145330494/0a101b8e-5f13-4936-b7a3-6b00595b7eb4)

**Comment by PabloGOP** [comment]
> > Sorry but I am not getting it, I think this repo has more commits than any other in the subject so I'd say that we crack at making commits...
> 
> i think that this is a great example to know how to make it:
> 
> ![Image](https://private-user-images.githubusercontent.com/145291757/324984099-7c2a7a82-f8c3-4b28-bb6a-9fb7548a2efd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDQ3ODksIm5iZiI6MTcxMzkwNDQ4OSwicGF0aCI6Ii8xNDUyOTE3NTcvMzI0OTg0MDk5LTdjMmE3YTgyLWY4YzMtNGIyOC1iYjZhLTlmYjc1NDhhMmVmZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMDM0NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYmU4YzJlZGUyZDgxMzcwY2YwZDg5ZDY5OWNjODA0ZGJiYjc4MGIxNzExYTVhYzliMWNjMTNjOTNmZWViODkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.81_UTo1h3XG5c4NbGaLSZjRwQocE2iVZCzUQCNg9mqA)

## I absolutely learnt from the best:
![image](https://github.com/Arquisoft/wiq_es04a/assets/113972208/87784bdf-77e9-4f98-a098-5cffacb1e671)

**Comment by PabloGOP** [comment]
Hold my beer @uo288245 

![image](https://github.com/Arquisoft/wiq_es04a/assets/113972208/ffb81cb0-10bc-43c6-8669-893652b01fdb)

**Comment by PBC003** [comment]
😯My favourite is that one:😯

![image](https://github.com/Arquisoft/wiq_es04a/assets/132336640/a2c3be6e-5201-4af1-ab37-b63df26437fe)

**Comment by uo288245** [comment]
> 😯My favourite is that one:😯
> 
> ![image](https://private-user-images.githubusercontent.com/132336640/324991839-a2c3be6e-5201-4af1-ab37-b63df26437fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDYwNzcsIm5iZiI6MTcxMzkwNTc3NywicGF0aCI6Ii8xMzIzMzY2NDAvMzI0OTkxODM5LWEyYzNiZTZlLTUyMDEtNGFmMS1hYjM3LWI2M2RmMjY0MzdmZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMDU2MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wODU4OThkYmI4YWZlYmY2MmY5NmQxZGU0Y2FhODEzODdjZmJiNzA2YjFhZWNlNjQ5ODQ2YjA0NTc0M2VmZGZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7j-eIgbhXfxpZlKMTPsE0-TL-CRRCL7FSpB0TgPT5t0)

Samu?

**Comment by PBC003** [comment]
No, I think this one is from Wiffo 😅😂

**Comment by uo289689** [comment]
![image](https://github.com/Arquisoft/wiq_es04a/assets/145044934/07bca4b9-3204-4e10-9f09-195a597d0cf4)
I believe you should practice what you preach; however, I think commit size is not relevant; what matters is the total contribution to the project.

**Comment by PBC003** [comment]
😈Welcome to pair programming😈

![image](https://github.com/Arquisoft/wiq_es04a/assets/132336640/c4dc80ec-38ee-466d-ab78-db016414dd6e)

**Comment by uo288245** [comment]
> ![image](https://private-user-images.githubusercontent.com/145044934/324997911-07bca4b9-3204-4e10-9f09-195a597d0cf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDgwNjYsIm5iZiI6MTcxMzkwNzc2NiwicGF0aCI6Ii8xNDUwNDQ5MzQvMzI0OTk3OTExLTA3YmNhNGI5LTMyMDQtNGUxMC05ZjA5LTE5NWE1OTdkMGNmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTI5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmE5NTE4ZWVlZGUwOWIzZDI5NTUxZTk4MWQwNzQwODAxMDI1ZjllOWUxOGI4NTdkNzMyNzAzNzNhYzQ4MWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V7BhXJP727CgJDPPIyqbcCCllQd_CvhdJY0_ifcbmus) I believe you should practice what you preach. However, I think commit size is not relevant, what matters is the total contribution to the project.

## I absolutely learnt from the best:
![imagen](https://github.com/Arquisoft/wiq_es04a/assets/145291757/b55ac41f-ba3e-4a53-be2d-f12df461314f)

**Comment by uo289689** [comment]
> > ![image](https://private-user-images.githubusercontent.com/145044934/324997911-07bca4b9-3204-4e10-9f09-195a597d0cf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDgwNjYsIm5iZiI6MTcxMzkwNzc2NiwicGF0aCI6Ii8xNDUwNDQ5MzQvMzI0OTk3OTExLTA3YmNhNGI5LTMyMDQtNGUxMC05ZjA5LTE5NWE1OTdkMGNmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTI5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmE5NTE4ZWVlZGUwOWIzZDI5NTUxZTk4MWQwNzQwODAxMDI1ZjllOWUxOGI4NTdkNzMyNzAzNzNhYzQ4MWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V7BhXJP727CgJDPPIyqbcCCllQd_CvhdJY0_ifcbmus) I believe you should practice what you preach. However, I think commit size is not relevant, what matters is the total contribution to the project.
> 
> ## I absolutely learnt from the best:
> ![imagen](https://private-user-images.githubusercontent.com/145291757/325001405-b55ac41f-ba3e-4a53-be2d-f12df461314f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDg2NDgsIm5iZiI6MTcxMzkwODM0OCwicGF0aCI6Ii8xNDUyOTE3NTcvMzI1MDAxNDA1LWI1NWFjNDFmLWJhM2UtNGE1My1iZTJkLWYxMmRmNDYxMzE0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTM5MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYjc3ODA5ZGJlOTJkZjZhOGI5YmI1YzY4MmE5NDcyZDQ2NmI5NGRkMWMxM2JkNjM5YzNhYmMwNDZiYmJhMDlhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eMgh2VV-Ai5cC7qtBwUkxpUVd-j598Dd0dhK76gj5fE)

Personally, I don't see the problem with a small commit that fixes a bug. Could you tell me the disadvantages of small commits?

**Comment by marco-qg** [comment]
> > > ![image](https://private-user-images.githubusercontent.com/145044934/324997911-07bca4b9-3204-4e10-9f09-195a597d0cf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDgwNjYsIm5iZiI6MTcxMzkwNzc2NiwicGF0aCI6Ii8xNDUwNDQ5MzQvMzI0OTk3OTExLTA3YmNhNGI5LTMyMDQtNGUxMC05ZjA5LTE5NWE1OTdkMGNmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTI5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmE5NTE4ZWVlZGUwOWIzZDI5NTUxZTk4MWQwNzQwODAxMDI1ZjllOWUxOGI4NTdkNzMyNzAzNzNhYzQ4MWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V7BhXJP727CgJDPPIyqbcCCllQd_CvhdJY0_ifcbmus) I believe you should practice what you preach. However, I think commit size is not relevant, what matters is the total contribution to the project.
> > 
> > 
> > ## I absolutely learnt from the best:
> > ![imagen](https://private-user-images.githubusercontent.com/145291757/325001405-b55ac41f-ba3e-4a53-be2d-f12df461314f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDg2NDgsIm5iZiI6MTcxMzkwODM0OCwicGF0aCI6Ii8xNDUyOTE3NTcvMzI1MDAxNDA1LWI1NWFjNDFmLWJhM2UtNGE1My1iZTJkLWYxMmRmNDYxMzE0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTM5MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYjc3ODA5ZGJlOTJkZjZhOGI5YmI1YzY4MmE5NDcyZDQ2NmI5NGRkMWMxM2JkNjM5YzNhYmMwNDZiYmJhMDlhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eMgh2VV-Ai5cC7qtBwUkxpUVd-j598Dd0dhK76gj5fE)
> 
> Personally, I don't see the problem with a small commit that fixes a bug. Could you tell me the disadvantages of small commits?

Animal, you destroyed him

**Comment by uo288245** [comment]
> > > > ![image](https://private-user-images.githubusercontent.com/145044934/324997911-07bca4b9-3204-4e10-9f09-195a597d0cf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDgwNjYsIm5iZiI6MTcxMzkwNzc2NiwicGF0aCI6Ii8xNDUwNDQ5MzQvMzI0OTk3OTExLTA3YmNhNGI5LTMyMDQtNGUxMC05ZjA5LTE5NWE1OTdkMGNmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTI5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmE5NTE4ZWVlZGUwOWIzZDI5NTUxZTk4MWQwNzQwODAxMDI1ZjllOWUxOGI4NTdkNzMyNzAzNzNhYzQ4MWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V7BhXJP727CgJDPPIyqbcCCllQd_CvhdJY0_ifcbmus) I believe you should practice what you preach. However, I think commit size is not relevant, what matters is the total contribution to the project.
> > > 
> > > 
> > > ## I absolutely learnt from the best:
> > > ![imagen](https://private-user-images.githubusercontent.com/145291757/325001405-b55ac41f-ba3e-4a53-be2d-f12df461314f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDg2NDgsIm5iZiI6MTcxMzkwODM0OCwicGF0aCI6Ii8xNDUyOTE3NTcvMzI1MDAxNDA1LWI1NWFjNDFmLWJhM2UtNGE1My1iZTJkLWYxMmRmNDYxMzE0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTM5MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYjc3ODA5ZGJlOTJkZjZhOGI5YmI1YzY4MmE5NDcyZDQ2NmI5NGRkMWMxM2JkNjM5YzNhYmMwNDZiYmJhMDlhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eMgh2VV-Ai5cC7qtBwUkxpUVd-j598Dd0dhK76gj5fE)
> > 
> > 
> > Personally, I don't see the problem with a small commit that fixes a bug. Could you tell me the disadvantages of small commits?
> 
> Animal, you destroyed him

Small commits offer many benefits, such as easier code review, better collaboration, and quicker identification of bugs. However, they're not without their downsides:

    Fragmented History: With excessively small commits, the commit history can become cluttered and difficult to follow, making it harder to understand the evolution of the codebase over time.

    Overhead: Too many small commits can lead to increased overhead, both in terms of storage space and in the time it takes to manage and review them. This can become especially burdensome in large projects with many contributors.

    Incomplete Features: Breaking down tasks into very small commits might mean that each commit doesn't represent a fully functional or meaningful change on its own. This can make it harder to revert changes or to understand the purpose of a particular commit without looking at surrounding commits.

    Merge Conflicts: Small commits can increase the likelihood of merge conflicts, especially if multiple developers are working on closely related parts of the codebase simultaneously. Resolving these conflicts can be time-consuming and disruptive.

    Lack of Context: Small commits may lack sufficient context or documentation to explain the reasoning behind the changes. This can make it more challenging for future developers to understand why certain decisions were made or why specific changes were implemented.

    Continuous Integration Overhead: If each small commit triggers a separate build or test run in a continuous integration (CI) pipeline, it can lead to excessive resource usage and longer feedback cycles, slowing down development.

    Difficulty in Rollbacks: While small commits make it easier to pinpoint where an issue was introduced, they can also make it more difficult to roll back changes if necessary, particularly if dependencies between commits are complex or if changes are spread across multiple commits.

In summary, while small commits offer advantages, such as granularity and easier code review, they can also introduce overhead and complexities that need to be managed effectively. Balancing the size of commits based on the specific needs and constraints of the project is essential for maintaining a healthy and efficient development workflow.

**Comment by uo288245** [comment]
> > > > > ![image](https://private-user-images.githubusercontent.com/145044934/324997911-07bca4b9-3204-4e10-9f09-195a597d0cf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDgwNjYsIm5iZiI6MTcxMzkwNzc2NiwicGF0aCI6Ii8xNDUwNDQ5MzQvMzI0OTk3OTExLTA3YmNhNGI5LTMyMDQtNGUxMC05ZjA5LTE5NWE1OTdkMGNmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTI5MjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmE5NTE4ZWVlZGUwOWIzZDI5NTUxZTk4MWQwNzQwODAxMDI1ZjllOWUxOGI4NTdkNzMyNzAzNzNhYzQ4MWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V7BhXJP727CgJDPPIyqbcCCllQd_CvhdJY0_ifcbmus) I believe you should practice what you preach. However, I think commit size is not relevant, what matters is the total contribution to the project.
> > > > 
> > > > 
> > > > ## I absolutely learnt from the best:
> > > > ![imagen](https://private-user-images.githubusercontent.com/145291757/325001405-b55ac41f-ba3e-4a53-be2d-f12df461314f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5MDg2NDgsIm5iZiI6MTcxMzkwODM0OCwicGF0aCI6Ii8xNDUyOTE3NTcvMzI1MDAxNDA1LWI1NWFjNDFmLWJhM2UtNGE1My1iZTJkLWYxMmRmNDYxMzE0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyM1QyMTM5MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYjc3ODA5ZGJlOTJkZjZhOGI5YmI1YzY4MmE5NDcyZDQ2NmI5NGRkMWMxM2JkNjM5YzNhYmMwNDZiYmJhMDlhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eMgh2VV-Ai5cC7qtBwUkxpUVd-j598Dd0dhK76gj5fE)
> > > 
> > > 
> > > Personally, I don't see the problem with a small commit that fixes a bug. Could you tell me the disadvantages of small commits?
> > 
> > 
> > Animal, you destroyed him

 Small commits offer many benefits, such as easier code review, better collaboration, and quicker identification of bugs. However, they're not without their downsides:
Fragmented History: With excessively small commits, the commit history can become cluttered and difficult to follow, making it harder to understand the evolution of the codebase over time.
Overhead: Too many small commits can lead to increased overhead, both in terms of storage space and in the time it takes to manage and review them. This can become especially burdensome in large projects with many contributors.
 Incomplete Features: Breaking down tasks into very small commits might mean that each commit doesn't represent a fully functional or meaningful change on its own. This can make it harder to revert changes or to understand the purpose of a particular commit without looking at surrounding commits.
Merge Conflicts: Small commits can increase the likelihood of merge conflicts, especially if multiple developers are working on closely related parts of the codebase simultaneously. Resolving these conflicts can be time-consuming and disruptive.
Lack of Context: Small commits may lack sufficient context or documentation to explain the reasoning behind the changes. This can make it more challenging for future developers to understand why certain decisions were made or why specific changes were implemented.
Continuous Integration Overhead: If each small commit triggers a separate build or test run in a continuous integration (CI) pipeline, it can lead to excessive resource usage and longer feedback cycles, slowing down development.
Difficulty in Rollbacks: While small commits make it easier to pinpoint where an issue was introduced, they can also make it more difficult to roll back changes if necessary, particularly if dependencies between commits are complex or if changes are spread across multiple commits.
In summary, while small commits offer advantages, such as granularity and easier code review, they can also introduce overhead and complexities that need to be managed effectively. Balancing the size of commits based on the specific needs and constraints of the project is essential for maintaining a healthy and efficient development workflow.

**Comment by CANCI0** [comment]
It is clear that none of you know how to do commits properly. If I were you, I would take a look at the excellent work being done by the 1a team in terms of git management. Attached is an example of our magnificence:
![Captura](https://github.com/Arquisoft/wiq_es04a/assets/92929497/7887bb1d-815d-4cad-8ba6-5946c71e9722)

**Comment by iyanfdezz** [comment]
Sit and learn, kids.
![Captura de pantalla 2024-04-25 180459](https://github.com/Arquisoft/wiq_es04a/assets/145439249/45020246-c039-4105-8ba0-5197d2737517)

**Comment by PabloGOP** [comment]
> No, I think this one is from Wiffo 😅😂

Wrong, I do not comment my code as it is fully readable

**Comment by PabloGOP** [comment]
Let's take an ultimate look to this masterpiece from [wiq_es4d](https://github.com/Arquisoft/wiq_es04d):

<img width="995" alt="image" src="https://github.com/Arquisoft/wiq_es04a/assets/113972208/a9081939-95c0-4fd0-b459-4faf14820d6e">
